sync_crawler/crawlers/ltn_crawler.py

```diff
@@ -0,0 +1,116 @@
+from collections.abc import Iterable
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from dataclasses import dataclass
+from datetime import datetime
+from itertools import count
+from typing import Optional
+
+import bs4
+import requests
+from dateutil.parser import parse
+
+from sync_crawler.crawlers.base_crawler import BaseCrawler
+from sync_crawler.models.news import News
+
+
+@dataclass
+class LtnNewsMetadata:
+    publish_time: datetime
+    category: Optional[str]
+    url: str
+
+
+class LtnCrawler(BaseCrawler):
+    media_name = "ltn"
+    metadata_api = "https://news.ltn.com.tw/ajax/breakingnews/all/{}"
+
+    def read(self, start_from: datetime) -> Iterable[News]:
+        for page in count(1, step=1):
+            try:
+                metadatas = list(self._fetch_metadata(page))
+            except Exception as e:
+                self.logger.error(
+                    f'Stop crawling, because of {type(e).__name__}: "{e}"'
+                )
+                break
+
+            if not metadatas or metadatas[0].publish_time < start_from:
+                self.logger.info("Finish crawling.")
+                break
+
+            with ThreadPoolExecutor() as executor:
+                metadatas = filter(lambda x: x.publish_time >= start_from, metadatas)
+
+                future_to_url = {
+                    executor.submit(self._crawl_news, metadata): metadata
+                    for metadata in metadatas
+                }
+                for future in as_completed(future_to_url):
+                    try:
+                        news = future.result()
+                    except Exception as e:
+                        self.logger.error(
+                            f'{future_to_url[future].url}: {type(e).__name__}: "{e}"'
+                        )
+                    else:
+                        yield news
+
+    def _crawl_news(self, metadata: LtnNewsMetadata) -> News:
+        response = requests.get(metadata.url, allow_redirects=False)
+        response.raise_for_status()
+        response.encoding = "utf-8"
+
+        soup = bs4.BeautifulSoup(response.text, "html.parser")
+
+        title = soup.select("[itemprop=articleBody] > h1")[0].text.strip()
```

Comment:
**Suggestion:** Handle the potential `IndexError` that can occur if the `soup.select` does not find any matching elements. [possible bug, importance: 8]
```suggestion
        title_elements = soup.select("[itemprop=articleBody] > h1")
        if not title_elements:
            raise ValueError("Title element not found.")
        title = title_elements[0].text.strip()
```